tools-local/tasks/mobile.tasks/AndroidAppBuilder/AndroidAppBuilder.csproj

```diff
@@ -15,10 +15,16 @@
     <PackageReference Include="Microsoft.Build.Framework" Version="$(RefOnlyMicrosoftBuildFrameworkVersion)" />
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(RefOnlyMicrosoftBuildTasksCoreVersion)" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(RefOnlyMicrosoftBuildUtilitiesCoreVersion)" />
+    <PackageReference Include="System.Reflection.MetadataLoadContext" Version="4.7.1" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="ApkBuilder.cs" />
     <Compile Include="AndroidAppBuilder.cs" />
+    <Compile Include="AssemblyResolver.cs" />
     <Compile Include="Utils.cs" />
   </ItemGroup>
+
+  <Target Name="PublishBuilder"
+          AfterTargets="Build"
+          DependsOnTargets="Publish" />
```

Comment:
Why does it need to depend on publish?

Reply:
we need to "publish" *AppBuilders in order to consume them from MSBuild correctly, without this trick it fails with `Couldn't load System.Reflection.MetadataLoadContext` in run time. So basically we need `System.Reflection.MetadataLoadContext.dll` next to *AppBuilder.dll 🙂 
I copied this trick from WasmAppBuilder:https://github.com/dotnet/runtime/blob/master/tools-local/tasks/mobile.tasks/WasmAppBuilder/WasmAppBuilder.csproj#L21-L23